include/deal.II/fe/mapping_q_cache.h

```diff
@@ -111,6 +112,38 @@ class MappingQCache : public MappingQGeneric<dim, spacedim>
                const typename Triangulation<dim, spacedim>::cell_iterator &)>
                &compute_points_on_cell);
 
+  /**
+   * Initialize the data cache by computing the mapping support points for all
+   * cells (on all levels) of the given triangulation and a given @p mapping
+   * and transforming these points via the function @p transformation_function.
+   *
+   * If @p is_displacement_function is set to true, the return value of the
+   * function is interpreted as relative deformation and the result is added
+   * to the original point.
+   *
+   * This function calls the above function so that the comments there regarding
+   * threading are also true here.
+   *
+   * @note The cache is invalidated upon the signal
+   * Triangulation::Signals::any_change of the underlying triangulation.
+   */
+  void
+  initialize(const Triangulation<dim, spacedim> &tria,
+             const Mapping<dim, spacedim> &      mapping,
+             const std::function<Point<spacedim>(
+               const typename Triangulation<dim, spacedim>::cell_iterator &,
+               const Point<spacedim> &)> &       transformation_function,
+             const bool is_displacement_function = false);
+
+  /**
+   * The same as above but taking a dealii::Function object.
+   */
+  void
+  initialize(const Triangulation<dim, spacedim> &tria,
+             const Mapping<dim, spacedim> &      mapping,
+             const Function<spacedim> &          transformation_function,
+             const bool is_displacement_function = false);
```

Comment:
Generally, I would prefer if the default setting is `true` independently of which logic is implemented as default. 

In ALE literature, the nomenclature is "mesh movement/motion", meaning displacements of the mesh as compared to a "undeformed" reference configuration (i.e. the mesh described by mapping). Hence, I would prefer something like `const bool function_describes_relative_displacement = true`. 

If you want to have the default state as is, I would write `const bool function_maps_to_absolute_coordinates = true`.

Reply:
I agree with the `function_describes_relative_displacement` name - it is very easy to get confused by whether the value that gets used is an absolute coordinate position solely determined by the function value (argument would be `false`) or a relative displacement against the position determined by the mapping, so being as explicit as possible both in the argument name and in the comment is important.